app/controllers/tests_controller.rb

```diff
@@ -28,14 +28,16 @@ def create
       test.save!
     end
 
-    render status: :created, json: test
-
     FillExpectedOutputJob.perform_later(test.id)
 
-    # Rerun all previous submissions on this new test case
-    if problem.submission_ids.size > 0
-      RunSubmissionsJob.perform_later(*problem.submission_ids)
+    # Rerun most recent submissions of each user on this new test case
+    newest_submission_by_user = problem.submissions.group(:user_id).maximum(:id).values
```

Comment:
Might be good to indicate in the variable name that this will be a list of submission ids.